diff --git a/django/db/backends/mysql/compiler.py b/django/db/backends/mysql/compiler.py
index 3682d6da05..d304704efb 100644
--- a/django/db/backends/mysql/compiler.py
+++ b/django/db/backends/mysql/compiler.py
@@ -15,6 +15,8 @@ class SQLInsertCompiler(compiler.SQLInsertCompiler, SQLCompiler):
 
 class SQLDeleteCompiler(compiler.SQLDeleteCompiler, SQLCompiler):
     def as_sql(self):
+        if not self.query.alias_map:
+            self.query.get_initial_alias()
         if self.connection.features.update_can_self_select or self.single_alias:
             return super().as_sql()
         # MySQL and MariaDB < 10.3.2 doesn't support deletion with a subquery
diff --git a/django/db/models/sql/compiler.py b/django/db/models/sql/compiler.py
index 29e33c075f..c655b65e45 100644
--- a/django/db/models/sql/compiler.py
+++ b/django/db/models/sql/compiler.py
@@ -1423,6 +1423,8 @@ class SQLDeleteCompiler(SQLCompiler):
         Create the SQL for this query. Return the SQL string and list of
         parameters.
         """
+        if not self.query.alias_map:
+            self.query.get_initial_alias()
         if self.single_alias:
             return self._as_sql(self.query)
         innerq = self.query.clone()
